packages/puppeteer-core/src/common/bidi/Page.ts

```diff
@@ -14,27 +14,76 @@
  * limitations under the License.
  */
 
-import {Page as PageBase} from '../../api/Page.js';
+import * as Bidi from 'chromium-bidi/lib/cjs/protocol/protocol.js';
+
+import {Page as PageBase, PageEmittedEvents} from '../../api/Page.js';
 import {stringifyFunction} from '../../util/Function.js';
+import {ConsoleMessage, ConsoleMessageLocation} from '../ConsoleMessage.js';
 import type {EvaluateFunc, HandleFor} from '../types.js';
 import {isString} from '../util.js';
 
 import {Connection} from './Connection.js';
 import {JSHandle} from './JSHandle.js';
 import {BidiSerializer} from './Serializer.js';
-import {Reference} from './types.js';
 
 /**
  * @internal
  */
 export class Page extends PageBase {
   #connection: Connection;
+  #subscribedEvents = [
+    'log.entryAdded',
+  ] as Bidi.Session.SubscribeParameters['events'];
   _contextId: string;
 
   constructor(connection: Connection, contextId: string) {
     super();
     this.#connection = connection;
     this._contextId = contextId;
+
+    this.connection.send('session.subscribe', {
+      events: this.#subscribedEvents,
+      contexts: [this._contextId],
+    });
+
+    this.connection.on('log.entryAdded', this.#onLogEntryAdded.bind(this));
```

Comment:
TODO: we need to unsubscribe at some point

Reply:
From my understanding we don't need to unsubscribe per BiDi. Also in Puppeteer we don't have a use case for it either (looking at CDP implementation).

Reply:
I don't think it will cause a memory leak, there are no should be no references stored on the BiDi side as all RemoteValue have `root: none` when being send. Do you think the connection can cause it?

Reply:
I don't think it has to do with the BiDi references. Let's chat on Monday to clarify this.